extensions/skyportal/skyportal/handlers/api/alert.py

```diff
@@ -245,18 +257,26 @@ def post_alert(
         raise ValueError(f"Failed to fetch data for {object_id} from Kowalski")
 
     if len(latest_alert_data) > 0:
+        latest_alert_data["candidate"]["candid"] = int(
+            latest_alert_data["candidate"]["candid"]
+        )
         candids = {a.get("candid", None) for a in alert_data["prv_candidates"]}
         if latest_alert_data["candidate"]["candid"] not in candids:
             alert_data["prv_candidates"].append(latest_alert_data["candidate"])
 
-    df = pd.DataFrame.from_records(alert_data["prv_candidates"])
-    mask_candid = df["candid"] == str(candid)
-
-    if candid is None or sum(mask_candid) == 0:
+    if candid is None:
         candid = int(latest_alert_data["candidate"]["candid"])
```

Comment:
what if `latest_alert_data` is an empty dict here?

Reply:
Good idea, adding an extra check there.